### PR TITLE
Api provides the ability for editor to publish articles

### PR DESCRIPTION
**Pull request by Cma-B**
### User Story
```
As an api
in order for editor to choose which article to show the users
I want to be able to publish articles.

```
## Changes proposed in this pull request:
* provide the ability for editor as a role to publish the articles.
* adds namespace as Admin to control all the actions from admin side
## What I have learned working on this feature:
* adding extra controller to have dry codes

### CODE REVIEW

**Comment by Odeane** [comment]
reviewed

**Review by Alex-Bellamy — APPROVED** [review]
looking good peeps

File: spec/requests/api/v1/admin/editor_can_publish_article_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  let!(:journalist) { create(:user, role: "journalist")}
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(journalist_credentials) }
+
+  let!(:article) { create(:article, category: 'sports', published: false ) }
+
+
+  describe "successfully" do
+
+  before do
```
Comment (oliverochman) [inline spec/requests/api/v1/admin/editor_can_publish_article_spec.rb]: please correct the indentation in this file

File: spec/requests/api/v1/admin/editor_can_publish_article_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    it "article should not be published" do
+      expect(Article.last["published"]).to eq false
+    end
+
+    it "should return error message" do
+      expect(response_json['message']).to eq "You are not authorized to access this action"
+    end
+  end
+end
```
Comment (oliverochman) [inline spec/requests/api/v1/admin/editor_can_publish_article_spec.rb:95]: what happens when someone that is not logged in tries to hit this route?

what happens if an editor tries to publish an article that already been published?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/admin/editor_can_publish_article_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  let(:editor_credentials) { editor.create_new_auth_token }
+  let(:editor_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(editor_credentials) }
+
+  let!(:journalist) { create(:user, role: "journalist")}
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(journalist_credentials) }
+
+  let!(:article) { create(:article, category: 'sports', published: false ) }
+
+
```
Comment (emtalen) [inline spec/requests/api/v1/admin/editor_can_publish_article_spec.rb:12]: Please format this file

File: app/controllers/api/v1/admin/articles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    render json: {message: 'You are not authorized to access this action'}, status: 401
+  end
+    
+  def attach_image(article)
+    params_image = params[:article][:image]
+      if params_image.present?
+        DecodeService.attach_image(params_image, article.image)
+      end
+    end
+  end
```
Comment (emtalen) [inline app/controllers/api/v1/admin/articles_controller.rb]: Please mind the indentation for the last ends here

**Review by emtalen — COMMENTED** [review]
don't forget the formatting!

File: app/controllers/api/v1/admin/articles_controller.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    
+  def authorize_journalist
+    unless current_user.role == 'journalist'
+      unauthorized()
+    end
+  end
+    
+  def authorize_editor
+    unless current_user.role == 'editor'
+      unauthorized()
```
Comment (oliverochman) [inline app/controllers/api/v1/admin/articles_controller.rb]: please remove the parenthesis here, not needed

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]
Good job!

**Pull request merged**